src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,22 +29,28 @@
 import javax.servlet.http.HttpServletResponse;
 import java.util.logging.Logger;
 
+/** Servlet that recieves user location and sends small business output to the client **/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  SmallCityService smallCityService;
+  SmallCityService smallCityService = new SmallCityService();
   private final static Logger LOGGER = Logger.getLogger(DataServlet.class.getName());
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
     String latString = request.getParameter("lat");
     String lngString = request.getParameter("lng");
     double lat = convertToDouble(latString);
     double lng = convertToDouble(lngString);
-    // Google Pittsburgh Office Location (hardcoded prototype)
-    // Will be populated with lat and lng doubles
-    MapLocation userLocation = new MapLocation(40.457410, -79.916573);
-    smallCityService = new SmallCityService(userLocation);
+    MapLocation userLocation = new MapLocation(lat, lng);
+    smallCityService.createUserWithGeolocation(userLocation);
+    } catch(NullPointerException e) {
+      String zip = request.getParameter("zipCode");
+      smallCityService.createUserWithZip(zip);
+      LOGGER.warning(e.getMessage());
+      LOGGER.warning("Unable to geolocate user, zipCode entered instead.");
```

Comment:
nit: Can this be converted to use string concatenation for just one log statement?

Reply:
Done!